mozilla/fx-private-relay-add-on
# [MPP-1708] Popup Refactor - News Panel

## Summary

Links:
- Jira Ticket: https://mozilla-hub.atlassian.net/browse/MPP-1769
- Epic PR #456 

TODO:
- [x] Dynamical build news list (based on list, audience criteria)
- [x] Dynamical build news detail (story) view
- [x] ~Add "unread' number to icon/logic to control it~
- [x] Add previous news items

Caveats:
- There is a bug where the order of the news items is not set but the content. I'll need to debug. 
- There's currently no logic set to only show to specific audiences or locales. (Everyone gets the same prompts)  

Acceptance Criteria: 
- [x] When there is a new feature show a yellow indicator with a count of the new features.
- [x] Once user has seen them all just show place holder screen
- [x] News and history area with toggle 
- [x] If a user click on a new item then we show a larger panel

## Testing

1. Open news item 
- Run add-on
- Log in
- Open panel > Click news
- **Expected:**  You should have two news items (Bundle, Phones) 
- Click on news item
- **Expected:** New item detail should open up
- Click back button 
- **Expected:** You should now be back at all news panel 

2. Add waffle flag, confirm news item exists 
- Add waffle flag to local Relay instance: `firefox_integration` _(Do this before running the add-on)_
- Run add-on
- Log in
- Open panel > Click news
- **Expected:**  You should have three news items (Bundle, Phones and Firefox Integration) 

## Screenshots

<img width="388" alt="image" src="https://user-images.githubusercontent.com/2692333/214765793-3b618123-43e5-4515-aafc-31960d2bc933.png">

<img width="387" alt="image" src="https://user-images.githubusercontent.com/2692333/214765837-4b729b5b-0062-4345-ae41-6dab63661190.png">
